### PR TITLE
Use kotlin 1.9

### PR DESCRIPTION
**Pull request by sstone**
This PR upgrades our code, build scripts and publishing scripts to use kotlin 1.9.
We also add an  `iosSimulatorArm64` target for users wishing to run tests against a simulator on Apple processors.

### CODE REVIEW

File: tests/build.gradle.kts [hunk 1]
```diff
@@ -36,32 +36,33 @@ kotlin {
...
+    iosSimulatorArm64()
 }
 
 val includeAndroid = System.getProperty("includeAndroid")?.toBoolean() ?: true
 if (includeAndroid) {
     extensions.configure<com.android.build.gradle.LibraryExtension>("android") {
         defaultConfig {
-            compileSdkVersion(30)
-            minSdkVersion(21)
+            compileSdk =30
```
Comment (t-bast) [inline tests/build.gradle.kts:64]: nit:

```suggestion
            compileSdk = 30
```

Comment (sstone) [reply tests/build.gradle.kts]: fixed in https://github.com/ACINQ/secp256k1-kmp/pull/92/commits/074c7df66e929fb3c47bafd1d3cf0a282816ea30

**Review by t-bast — APPROVED** [review]

**Comment by sstone** [comment]
This PR has been tested with `bitcoin-kmp` (see https://github.com/ACINQ/bitcoin-kmp/pull/108)  and `lightning-kmp` ( see https://github.com/ACINQ/lightning-kmp/pull/565),  AFAICT it's ready to be merged but I'll keep it open until we've tested it with `Phoenix`.

**Review by dpad85 — APPROVED** [review]
Successfully tested on Phoenix with Kotlin 1.9.
See https://github.com/ACINQ/phoenix/pull/509

**Pull request merged**